### ISSUE TITLE
Images are not showing up when it behinds proxy

### ISSUE THREAD

**Issue by dbaron92** [sahara101/Movie-Roulette#14]
I noticed that images (poster and background) are not showing up when using a domain. I used a domain through my NPM (Nginx Proxy Manager) as reverse proxy and points to http://server-ip:4000. However, I can see movie details and filter works. If I am using server IP directly, everything works as expected. I also tried AdGuard's DNS rewrite and it showed same issue.

I tried both instances (docker and ENV). Both shows same result.
![Screenshot 2024-12-10 170451](https://github.com/user-attachments/assets/b492a2e9-0e28-4e4b-ad54-e3c668c922e2)

**Comment by sahara101** [comment]
Hello. It is a CORS issue, as per troubleshoting:

Issue: Browser doesn’t load the poster or background.

Use the FQDN (Fully Qualified Domain Name) for Plex/Jellyfin in the environment variables/settings instead of the IP address.

Do you have both the Movie Roulette and Plex with FQDN?

**Comment by dbaron92** [comment]
For Plex URL, I used looks like this
https://192-168-1-X.xxxxxxxxxxxxxxxxxxx.plex.direct:32400
Im wondering if plex.direct blocks CORS.

**Comment by sahara101** [comment]
Could be, can you try the plex NPM url?
I also see the filter is cut off, tell me after the pictures work if it is still cut off and on what monitor resolution/device you have the issue.

**Comment by dbaron92** [comment]
I tried plex.example.com and movie roulette's setting didnt work due to "Failed to fetch libraries" error after clicking fetch libraries button. I had same issue when I used IP like https://192.168.1.X:32400. My plex server's "Secure connections" is "Required" not "Preferred". I dont know if it caused the problem.

My NPM proxy host is https://192.168.1.X:32400 and it loads up with XML file because I didnt add "/web" in NPM advanced setting.

**Comment by sahara101** [comment]
Try this: 

Go to settings → Network and activate Enable Local Network Discovery (GDM).

I also have NPM and have both plex and roulette in it, no special settings 

![image](https://github.com/user-attachments/assets/eabb3e67-66cb-4141-99d2-ef82e30959fd)

**Comment by dbaron92** [comment]
Yep its already checked for GDM and same settings in NPM except HTTPS. Im curious.. are you using required or preferred for connection in plex settings? I changed it to preferred and unchecked " Strict TLS configuration" and no luck. I might just setup a dummy plex server with default settings to see my domain will work with it.

**Comment by sahara101** [comment]
I have it preferred and strict TLS disabled. Try with http in NPM please

**Comment by dbaron92** [comment]
Yeah I tried http earlier and it didnt work. So I decided to built a secondary plex server and 2nd movie roulette server (port 4001). I tried http my domain name and it worked fine on secondary server's web app.
I tried that domain url on roulette's 4001 and I got same error "Failed to fetch libraries". I looked it up in F12 and its 500 server internal error and it came from roulette's IP. Do we have any log locations to look and find the issue?

**Comment by dbaron92** [comment]
I have some updates. I kept messing with docker and I realized that I had ENV setup. So I went to ENV and updated Plex info with secondary server. Everything works great now with ENV even the info is exactly same that I tried on docker build. Its so strange

I noticed that the cache gets lost every time when I stopped the python script or reboot the host. It doesn't bother me bec it will take some time to startup like few mins to get new cache.

You mentioned about filter was cut off. Im using Chrome on windows pc, everything shows up, and filter is still cut off
![Screenshot 2024-12-10 215011](https://github.com/user-attachments/assets/cb77444e-c153-451a-9e45-4952b5876366)

**Comment by sahara101** [comment]
I do not fully understand your last message. Is it now working with docker and only the new plex server? How is the seconds plex server different from the first one? Can you give me the inspect logs and the docker logs please. 

The cache must be persistent, does the local files get created? Can you show me the compose you used. 

As for the filter cut off, I will make a small update to see if it helps.

**Comment by dbaron92** [comment]
Sorry for the confusion. English isnt my first language :) 

I figured it out..
After looking a docker.log, I found the problem.. My cloudflare blocked it because I didnt realized that I placed the container in my main Docker server that is behind VPN that's connect to a different country for Plex arr stuff. So my cloudflare's WAF rule blocks any requests if its not from my country lol. 
So I made a new installation on different LXC/host and my custom domain is working good now in docker container. That's why my ENV always worked with my custom domain because its on a different host and not on VPN.

For docker local files, I can see them in docker's directory. For ENV, I don't see any changes and kept creating cache every time when python starts. I just rerun python script, save output into script.log, and redacted some sensitive info in it..  The log is here - [script.log.txt](https://github.com/user-attachments/files/18098231/script.log.txt)

I downloaded this project from this [https://github.com/sahara101/Movie-Roulette/releases/tag/v3.1](url).


By the way, I would like to hide settings in Docker version. I saw your another post that it is for ENV version - DISABLE_SETTINGS: TRUE.. Im wondering if we can use like this for Docker compose file:
    environment:
      - DISABLE_SETTINGS: TRUE
